src/shadowbox/server/manager_service.ts

```diff
@@ -146,6 +152,42 @@ export class ShadowsocksManagerService {
     }
   }
 
+  // Sets the default ports for new access keys
+  public async setPortForNewAccessKeys(req: RequestType, res: ResponseType, next: restify.Next):
+      Promise<void> {
+    const invalidPortArgument = (message: string) => new restify.BadRequestError(message);
+    try {
+      logging.debug(`setPort[ForNewAccessKeys request ${JSON.stringify(req.params)}`);
+      if (!req.params.port) {
+        return next(
+            invalidPortArgument(`Expected a port argument but found none. Request: ${req}`));
+      }
+
+      const port = req.params.port;
+      if (typeof port !== 'number') {
+        return next(invalidPortArgument(`Expected an numeric port, instead got ${port}`));
+      }
+      if (port < 1 || port > 65535) {
```

Comment:
You can delete this check. Other than checking the port is of number type, you can leave the rest of the checks to the setPortForNewAccessKeys call. The new typed errors is great for that.